slashing-protection/referencetests/src/test/java/tech/pegasys/web3signer/slashingprotection/model/AttestionTestModel.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.web3signer.slashingprotection.model;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.units.bigints.UInt64;
+
+public class AttestionTestModel {
+
+  private final Bytes publickKey;
+  private final UInt64 sourceEpoch;
+  private final UInt64 targetEpoch;
+  private final boolean shouldSucceed;
+  private final Bytes signingRoot;
+
+  public AttestionTestModel(
+      @JsonProperty(value = "pubkey", required = true) final Bytes publickKey,
+      @JsonProperty(value = "source_epoch", required = true) UInt64 sourceEpoch,
+      @JsonProperty(value = "target_epoch", required = true) UInt64 targetEpoch,
+      @JsonProperty(value = "should_succeed", required = true) boolean shouldSucceed,
+      @JsonProperty(value = "signing_root") Bytes signingRoot) {
```

Comment:
Should this be required too?

Reply:
no - the reference test files don't always define a signing_root - in fact, they may never ... but I would argue the tests SHOULD always have one, but that's just not the case.

Reply:
so, there's a bit of work going on with the EIP and the ref tests - so may get an update which makes this mandatory